test/integration/hotrestart_test.sh

```diff
@@ -152,13 +153,15 @@ function run_testsuite() {
 
   enableHeapCheck
 
+  # For SERVER_1, we set a drain timer, so it exits a few seconds after SERVER_2 starts.
   start_test Starting epoch 1
   ADMIN_ADDRESS_PATH_1="${TEST_TMPDIR}"/admin.1."${TEST_INDEX}".address
   run_in_background_saving_pid "${ENVOY_BIN}" -c "${UPDATED_HOT_RESTART_JSON}" \
       --restart-epoch 1 --base-id "${BASE_ID}" --service-cluster cluster --service-node node \
-      --use-fake-symbol-table "$FAKE_SYMBOL_TABLE" --admin-address-path "${ADMIN_ADDRESS_PATH_1}"
+      --use-fake-symbol-table "$FAKE_SYMBOL_TABLE" --admin-address-path "${ADMIN_ADDRESS_PATH_1}" \
+      --drain-time-s 2
```

Comment:
FWIW I don't think this actually impacts your test, since parent-shutdown-time is really what causes the exit. This just governs the drain close ramp time which doesn't matter in this case. Maybe remove this just to make the test more clear? Same in comments below if you agree.

Reply:
agreed; will remove.